src/installer/installer.go

```diff
@@ -612,7 +612,7 @@ func (i *installer) updateReadyMasters(nodes *v1.NodeList, readyMasters *[]strin
 				return fmt.Errorf("Node %s is not in inventory hosts", node.Name)
 			}
 			ctx = utils.GenerateRequestContext()
-			if err := i.inventoryClient.UpdateHostInstallProgress(ctx, host.Host.ID.String(), models.HostStageJoined, ""); err != nil {
+			if err := i.inventoryClient.UpdateHostInstallProgress(ctx, host.Host.InfraEnvID.String(), host.Host.ID.String(), models.HostStageJoined, ""); err != nil {
```

Comment:
```suggestion
			if err := i.inventoryClient.UpdateHostInstallProgress(ctx, config.InfraEnvID, host.Host.ID.String(), models.HostStageJoined, ""); err != nil {
```

Reply:
This is not a correct change - we are updating the status of the other masters, while running on bootstrap node, so we need the other hosts InfraEnvId, and not the InfraEnv ID of the bootstrap node

Reply:
Is it possible that they will have a different infra env???

Reply:
Perhaps I'm missing what does InfraEnv stands for?

Reply:
You can construct a cluster from hosts that are coming from different InfraEnvs (pools)